mesonbuild/modules/wayland.py

```diff
@@ -0,0 +1,115 @@
+# Copyright © 2020 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import typing as T
+from pathlib import PurePath
+
+from . import ExtensionModule
+from . import ModuleReturnValue
+from .. import build
+from .. import mlog
+from ..interpreterbase import permittedKwargs, stringArgs
+from ..mesonlib import MesonException, File, MachineChoice, version_compare
+
+if T.TYPE_CHECKING:
+    from ..interpreter import ModuleState, Interpreter
+
+
+class WaylandModule(ExtensionModule):
+
+    def __init__(self, interpreter: 'Interpreter'):
+        super().__init__(interpreter)
+
+        # wayland-scanner has a pkg-config file, we should use that to find
+        # wayland-scanner, as it's more robust
+        scanner_dep  = interpreter.dependency_impl(
+            'wayland-scanner', 'wayland-scanner', {'required': True, 'native': True})
+        scanner_name = scanner_dep.variable_method([], {'pkgconfig': 'wayland_scanner'})
+        self.scanner = interpreter.find_program_impl(
+            [scanner_name], silent=False, for_machine=MachineChoice.BUILD)
+
+        # Older version of wayland-scanner take different arguments
+        self.old_scanner = version_compare(scanner_dep.held_object.version, '< 1.15')
+
+        # We need this to find the location where upstream protocols are installed
+        self.protocols_dep = interpreter.dependency_impl(
+            'wayland-protocols', 'wayland-protocols', {'required': True})
+
+    @stringArgs
+    @permittedKwargs(build.CustomTarget.known_kwargs | {'is_protocol'})
```

Comment:
Do we really want to expose every current and future kwargs of custom_target()? Do they all make sense? Maybe you whould copy your own list here.